### PR TITLE
Move Cartesian and Pkg docs inline.

### PR DESCRIPTION
**Pull request by MichaelHatherly**
Also adds docs for `@ncall`, `@nextract`, and `@nany` which were exported but undocumented. /cc @timholy could you have a look at the additions if you have a chance to? Thanks.

### CODE REVIEW

File: base/cartesian.jl [hunk 1]
```diff
@@ -6,7 +6,35 @@ export @nloops, @nref, @ncall, @nexprs, @nextract, @nall, @nany, @ntuple, @nif
...
+    for i_2 = 1:size(A, 2)
+        j_2 = min(i_2, 5)
+        for i_1 = 1:size(A, 1)
+            j_1 = min(i_1, 5)
+            s += A[j_1,j_2]
+        end
+    end
+
+If you want just a post-expression, supply `nothing` for the pre-expression. Using
+parenthesis and semicolons, you can supply multi-statement expressions.
```
Comment (tkelman) [inline base/cartesian.jl]: parentheses

File: base/pkg.jl [hunk 2]
```diff
@@ -19,6 +19,37 @@ end
 const cd = Dir.cd
 
 dir(path...) = Dir.path(path...)
+
+"""
+    dir() -> AbstractString
+
+Returns the absolute path of the package directory. This defaults to
+`joinpath(homedir(),".julia","v\$(VERSION.major).\$(VERSION.minor)")` on all platforms (i.e.
+`~/.julia/v0.4` in UNIX shell syntax). If the `JULIA_PKGDIR` environment variable is set,
```
Comment (tkelman) [inline base/pkg.jl]: maybe this line should be interpolated so this doesn't have to manually be updated each new version?

File: base/cartesian.jl [hunk 3]
```diff
@@ -6,7 +6,35 @@ export @nloops, @nref, @ncall, @nexprs, @nextract, @nall, @nany, @ntuple, @nif
...
+    @nloops N itersym rangeexpr bodyexpr
+    @nloops N itersym rangeexpr preexpr bodyexpr
+    @nloops N itersym rangeexpr preexpr postexpr bodyexpr
+
+Generate `N` nested loops, using `itersym` as the prefix for the iteration variables.
+`rangeexpr` may be an anonymous-function expression, or a simple symbol `var` in which case
+the range is `1:size(var,d)` for dimension `d`.
+
+Optionally, you can provide "pre" and "post" expressions. These get executed first and last,
+respectively, in the body of each loop. For example, :
```
Comment (timholy) [inline base/cartesian.jl]: `, :`

**Comment by timholy** [comment]
LGTM. Thanks for doing this!

`@nextract` wasn't exported because you can do it almost as easily with `@nexpr`, but `@nextract` was older so I exported it for compatibility. However, we haven't rushed to ditch `@nextract`, so better to have it documented. Thanks again.

File: base/pkg.jl [hunk 4]
```diff
@@ -26,37 +57,174 @@ function __init__()
...
+
+If `pkg` has a URL registered in `Pkg.dir("METADATA")`, clone it from that URL on the
+default branch. The package does not need to have any registered versions.
+"""
 clone(url_or_pkg::AbstractString) = cd(Entry.clone,url_or_pkg)
+
+"""
+    clone(url, [pkg])
+
+Clone a package directly from the git URL `url`. The package does not need to be a
```
Comment (tkelman) [inline base/pkg.jl]: need to be ~~a~~

File: base/pkg.jl [hunk 5]
```diff
@@ -26,37 +57,174 @@ function __init__()
...
+"""
 clone(url::AbstractString, pkg::AbstractString) = cd(Entry.clone,url,pkg)
 
+"""
+    checkout(pkg, [branch="master"]; merge=true, pull=true)
+
+Checkout the `Pkg.dir(pkg)` repo to the branch `branch`. Defaults to checking out the
+"master" branch. To go back to using the newest compatible released version, use
+`Pkg.free(pkg)`. Changes are merged (fast-forward only) if the keyword argument `merge ==
+true`, and the latest version is pulled from the upsream repo if `pull == true`.
```
Comment (tkelman) [inline base/pkg.jl]: upstream

File: base/pkg.jl [hunk 6]
```diff
@@ -26,37 +57,174 @@ function __init__()
...
+    pin(pkg, version)
+
+Pin `pkg` at registered version `version`.
+"""
 pin(pkg::AbstractString, ver::VersionNumber) = cd(Entry.pin,pkg,ver)
 
+"""
+    update()
+
+Update package the metadata repo – kept in `Pkg.dir("METADATA")` – then update any fixed
```
Comment (tkelman) [inline base/pkg.jl]: Update ~~package~~ the metadata repo

**Comment by MichaelHatherly** [comment]
Thanks for reviewing, updated.

**Pull request merged**